test/integration/utils.ts

```diff
@@ -0,0 +1,307 @@
+    this.app = await appFactory(appOptions)
+
+    // Start server
+    const port = config.get('port')
+    const server = this.app.app.listen(port)
+    this.logger.info('Cache service started')
+
+    server.on('listening', () =>
+      this.logger.info(`Server started on port ${port}`)
+    )
+
+    process.on('unhandledRejection', err =>
+      this.logger.error(`Unhandled Rejection at: ${err}`)
+    )
+  }
+
+  async stop (): Promise<void> {
+    if (this.app) {
+      this.app.stop()
+    }
+
+    await this.sequelize?.close()
+    resetStore()
+
+    this.sequelize = undefined
+    this.app = undefined
+    this.eth = undefined
+    this.storageContract = undefined
+    this.consumerAddress = ''
+    this.providerAddress = ''
+  }
+
+  private async purgeDb (): Promise<void> {
+    try {
+      await fs
+        .unlink(path.join(process.cwd(), config.get<string>('db')))
+    } catch (e) {
+      // File does not exist
+      if (e.code !== 'ENOENT') {
+        throw e
+      }
+    }
+  }
+
+  private async initBlockchainProvider (): Promise<void> {
+    this.eth = ethFactory()
+    const [provider, consumer, ...accounts] = await this.eth.getAccounts()
+    this.providerAddress = provider
+    this.consumerAddress = consumer
+    this.accounts = accounts
+  }
+
+  public getRandomAccount () {
+    const acc = this.accounts[this.nextAccountIndex]
+
+    if (!acc) {
+      throw new Error(`Are are used all of yours ${this.nextAccountIndex - 1} accounts`)
+    }
+    this.nextAccountIndex += 1
+    return acc
+  }
+
+  private async deployStorageManager (): Promise<void> {
+    if (!this.eth || !this.providerAddress) {
+      throw new Error('Provider should be initialized and has at least 2 accounts')
+    }
+    const contract = new this.eth.Contract(storageManagerContract.abi as AbiItem[])
+    const deploy = await contract.deploy({ data: storageManagerContract.bytecode })
+    this.storageContract = await deploy.send({ from: this.providerAddress, gas: await deploy.estimateGas() })
+    await this.storageContract?.methods.setWhitelistedTokens(ZERO_ADDRESS, true).send({ from: this.providerAddress })
+  }
+
+  private async deployStaking (): Promise<void> {
+    if (!this.eth || !this.providerAddress) {
+      throw new Error('Provider should be initialized and has at least 2 accounts')
+    }
+    const contract = new this.eth.Contract(stakingContract.abi as AbiItem[])
+    const deploy = await contract.deploy({ arguments: [this.storageContract?.options.address], data: stakingContract.bytecode })
+    this.stakingContract = await deploy.send({ from: this.providerAddress, gas: await deploy.estimateGas() })
+    await this.stakingContract?.methods.setWhitelistedTokens(ZERO_ADDRESS, true).send({ from: this.providerAddress })
+  }
+
+  public async createOffer (offerData: Record<any, any>) {
+    const setOffer = this.storageContract?.methods.setOffer(
+      offerData.totalCapacity,
+      [offerData.periods],
+      [offerData.prices],
```

Comment:
Hmm why do you wrap it in arrays? If I saw correctly you are already passing arrays in tests...

Reply:
This is related to multicurrency changes. Now we had periods: `Array<Array<number>>` where the index of parent array mapped with a token on the same index in tokens array